src/DotVVM.Framework.Tests.Common/Binding/BindingCompilationTests.cs

```diff
@@ -145,6 +145,62 @@ public void BindingCompiler_Valid_MethodCall()
             Assert.AreEqual(viewModel.StringProp, "hulabula13");
         }
 
+        [TestMethod]
+        public void BindingCompiler_Valid_Lambda_Test()
+        {
+            var viewModel = new TestViewModel { StringProp = "abc" };
+            var function1 = (Func<string, string>)ExecuteBinding("(string arg) => SetStringProp(arg, 123)", viewModel);
+            var result1 = function1("test");
+            Assert.AreEqual("abc", result1);
+            Assert.AreEqual("test123", viewModel.StringProp);
+
+            var function2 = (Func<char, int>)ExecuteBinding("(char arg) => GetCharCode(arg)", viewModel);
+            var result2 = function2('A');
+            Assert.AreEqual(65, result2);
+        }
+
+        [TestMethod]
+        [DataRow("() => ;", typeof(Action), null)]
+        [DataRow("() => \"HelloWorld\"", typeof(Func<string>), typeof(string))]
+        [DataRow("() => 11", typeof(Func<int>), typeof(int))]
+        [DataRow("() => 4.5f", typeof(Func<float>), typeof(float))]
+        [DataRow("() => 4.5", typeof(Func<double>), typeof(double))]
+        [DataRow("() => 11 + 4.5", typeof(Func<double>), typeof(double))]
+        [DataRow("() => new System.Collections.Generic.List<int>()", typeof(Func<List<int>>), typeof(List<int>))]
```

Comment:
Parser currently does not support the `new` operator. Due to this fact, similar constructs currently fail. Should we support this? I think that someone could try to write something similar.

Reply:
Thanks to the type metadata in #905, we could support this in the future (with some limitations like default property values).